planner/core/preprocess.go

```diff
@@ -1592,11 +1592,12 @@ func (p *preprocessor) handleAsOfAndReadTS(node *ast.AsOfClause) {
 		return
 	}
 	if p.LastSnapshotTS != 0 {
-		dom := domain.GetDomain(p.ctx)
-		p.InfoSchema, p.err = dom.GetSnapshotInfoSchema(p.LastSnapshotTS)
+		is, err := p.ctx.GetSnapshotInfoSchema(p.LastSnapshotTS)
```

Comment:
Is this the only place that needs to get the snapshot infoschema with the temporary table?  If so, I prefer manually use 
```
&infoschema.TemporaryTableAttachedInfoSchema{
		InfoSchema:           is,
        LocalTemporaryTables: local.(*infoschema.LocalTemporaryTables)
}
```
instead of expose a new method to the `sessionctx.Context`.

> The bigger the interface, the weaker the abstraction – Rob Pike